dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -4941,7 +4941,8 @@ ReturnedColumn* buildAggregateColumn(Item* item, gp_walk_info& gwi)
         {
             for (uint32_t i = 0; i < gwi.returnedCols.size(); i++)
             {
-                if (*ac == gwi.returnedCols[i].get())
+                if (*ac == gwi.returnedCols[i].get() 
+                        && ac->alias() == gwi.returnedCols[i].get()->alias())
```

Comment:
Could you double check the case where projection column alias is empty, e.g. select sum(...) as `` ...?

Reply:
Sorts correctly when the sorting column is x or y and the other is ``. It does not sort by a column that has an empty column alias (existing behavior before chnage).